### PR TITLE
Add auth and comparison functions

### PR DESCRIPTION
**Pull request by freels**

### CODE REVIEW

File: test/query_test.rb [hunk 1]
```diff
@@ -363,6 +378,22 @@ def test_modulo
     assert_equal 2, client.query { modulo(2) }
   end
 
+  def test_lt
+    assert_equal true, client.query { lt 1, 2 }
```
Comment (ashfire908) [inline test/query_test.rb]: Assert both a true and false condition for these?

Comment (freels) [inline test/query_test.rb]: We only need to assert enough that the form is being submitted correctly. Remember we don't need to test the DB's behavior in any of these tests.

Comment (freels) [inline test/query_test.rb]: (At least, behavior doesn't matter unless the client makes some assumption about it.)

Comment (ashfire908) [inline test/query_test.rb]: Alright.

**Comment by ashfire908** [comment]
:sheep:

**Pull request merged**